recipes-soc/rdk-gstreamer-utils-platform/rdk-gstreamer-utils-platform.bb

```diff
@@ -0,0 +1,31 @@
+LICENSE = "CLOSED"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=175792518e4ac015ab6696d16c4f607e"
+
+DEPENDS += "${@bb.utils.contains('DISTRO_FEATURES', 'gstreamer1', 'gstreamer1.0  gstreamer1.0-plugins-base', 'gstreamer gst-plugins-base', d)}"
+DEPENDS += "rdk-gstreamer-utils-headers"
+
+SRC_URI = "git://git@github.com/ssadha725_comcast/rdk-gstreamer-utils-raspberrypi;${RDKE_GITHUB_SRC_URI_SUFFIX}"
+SRCREV = "a03aeef1c5978ab3c765307218f1b7b7c25616dd"
+S = "${WORKDIR}/git/"
+
+
+CXXFLAGS += "-I${STAGING_INCDIR}/glib-2.0 -I${STAGING_INCDIR}/gstreamer-1.0 -I${STAGING_DIR_TARGET}/${libdir}/glib-2.0/include/ "
+
+inherit autotools pkgconfig features_check
```

Comment:
what is features_check

Reply:
features_check class is used to ensure that certain features are available in the build environment before proceeding with the build. It allows recipes to specify dependencies on specific features, and if those features are not present, the build will fail with a clear error message.
We have conditional dependencies based on
 
DISTRO_FEATURES
DEPENDS += "${@bb.utils.contains('DISTRO_FEATURES', 'gstreamer1', 'gstreamer1.0  gstreamer1.0-plugins-base', 'gstreamer gst-plugins-base', d)}"

1. features_check ensures that the necessary features(gstreamer1) are enabled.
2. If required feature is missing, features-check provides a clear error message.
It is not strictly necessary in recipe.